### PR TITLE
input-ligand-smile

### PR DESCRIPTION
**Pull request by xiangchenjhu**

### CODE REVIEW

**Review by amitschang — COMMENTED** [review]
a couple of comments

**Comment by xiangchenjhu** [comment]
Goal:
Add a new feature to the pipeline that allows for the direct input of ligand SMILEs, providing a third input method for the pipeline.

File: bluephos/bluephos_pipeline.py [hunk 1]
```diff
@@ -70,16 +70,29 @@ def rerun_candidate_generator(input_dir, t_nn, t_ste):
...
 
-def get_generator(halides, acids, input_dir, t_nn, t_ste):
+def ligand_smiles_reader_generator(ligand_smiles):
+    ligand_df = pd.read_csv(ligand_smiles)
+    for _, row in ligand_df.iterrows():
+        ligand = {
+            "ligand_identifier": row["ligand_identifier"],
+            "ligand_SMILES": row["ligand_SMILES"],
+        }
+        yield pd.DataFrame([ligand])
```
[the remark this answers is not in the document]
Comment (xiangchenjhu) [reply bluephos/bluephos_pipeline.py:80]: Cool. That's more compact and flexible!

File: bluephos/tasks/nn.py [hunk 2]
```diff
@@ -107,6 +107,8 @@ def apply_nn(feature_df: pd.DataFrame, model_weights) -> pd.DataFrame:
 
 
 def nn(df: pd.DataFrame, element_features, train_stats, model_weights) -> pd.DataFrame:
+    if "z" not in df.columns:
+        df["z"] = None
```
[the remark this answers is not in the document]
Comment (xiangchenjhu) [reply bluephos/tasks/nn.py:111]: Yes, That's true! and I also updated the unnecessary initialization steps in the first task.

File: bluephos/bluephos_pipeline.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (xiangchenjhu) [reply bluephos/bluephos_pipeline.py]: Thanks for the review, updated based on it.

**Comment by xiangchenjhu** [comment]
Thanks for the review and updated based on it

File: bluephos/bluephos_pipeline.py [hunk 3]
```diff
@@ -58,8 +59,11 @@ def rerun_candidate_generator(input_dir, t_nn, t_ste):
         DataFrame: A single-row DataFrame containing candidate data.
     """
     for file in Path(input_dir).glob("*.parquet"):
+        print(file.name)
```
Comment (amitschang) [inline bluephos/bluephos_pipeline.py:62]: if you want to keep this, should probably make it a logger message

**Review by amitschang — COMMENTED** [review]
one small comment

File: bluephos/bluephos_pipeline.py (hunk 3 above)
Comment (xiangchenjhu) [reply bluephos/bluephos_pipeline.py]: That's the temporary code and should be removed. Thanks!

**Review by amitschang — APPROVED** [review]
Looks good, thanks 😄

**Pull request merged**